src/nucleus/tt.ml

```diff
@@ -135,10 +133,12 @@ and instantiate es ?(lvl=0) ({term=e';assumptions;loc;} as e) =
   in
   match e' with
 
-    | Type -> {e with assumptions}
+    | Type -> {e with assumptions} (* XXX why do we need assumptions here? Type is well formed by itself *)
```

Comment:
Sorry, I don't get that. In this `match` we discover that `e` is equal to `Type` (not that `e` has type  `t` which is equal to `Type`). It is *always* safe to drop assumptions on `Type` as it is well formed.